### PR TITLE
CueControl: Use mixxx::audio::FramePos internally

### PR DESCRIPTION
**Pull request by Holzhaus**

### CODE REVIEW

**Review by uklotzde — CHANGES_REQUESTED** [review]
That was heavy. Found only one leftover debugging statement.

@daschuer Please double check

File: src/engine/controls/cuecontrol.cpp [hunk 1]
```diff
@@ -640,24 +641,28 @@ void CueControl::loadCuesFromTrack() {
...
-                Cue::kNoPosition);
+                mainCuePosition,
+                mixxx::audio::kInvalidFramePos);
     }
-    m_pCuePoint->set(quantizeCuePoint(mainCuePoint.getPosition()));
+
+    DEBUG_ASSERT(mainCuePosition.isValid());
+    const auto quantizedMainCuePosition = quantizeCuePoint(mainCuePosition);
+    m_pCuePoint->set(quantizedMainCuePosition.toEngineSamplePosMaybeInvalid());
+    qWarning() << m_pCuePoint->get();
```
[the remark this answers is not in the document]
Comment (Holzhaus) [reply src/engine/controls/cuecontrol.cpp]: b8e4da11ac9400a9615f37ac29e272ba6aac039d

File: src/engine/controls/cuecontrol.cpp [hunk 2]
```diff
@@ -1089,51 +1106,68 @@ void CueControl::hotcueClear(HotcueControl* pControl, double value) {
...
         }
     }
 }
 
 void CueControl::hintReader(HintVector* pHintList) {
     Hint cue_hint;
-    double cuePoint = m_pCuePoint->get();
-    if (cuePoint >= 0) {
-        cue_hint.frame = SampleUtil::floorPlayPosToFrame(m_pCuePoint->get());
+    const auto cuePoint =
```
[the remark this answers is not in the document]
Comment (Holzhaus) [reply src/engine/controls/cuecontrol.cpp]: 2b559050c95f4b3e7d5c7404459b2d4c2eb563c6

**Comment by Holzhaus** [comment]
> That was heavy.

Yeah sorry. Maybe I should have migrated the individual methods in separate commits/PRs.

**Review by uklotzde — APPROVED** [review]

**Comment by daschuer** [comment]
LGTM. 
I think that deserves also a manual tests.
Did you give it a try already? I can test tonight.

**Comment by uklotzde** [comment]
@daschuer Ping. This is the last pending PR before the final cleanup.

**Review by daschuer — APPROVED** [review]
I did a brief test and everything still works. Thank you.

**Pull request merged**